mysql-test/suite/innodb/t/temp_table_encrypt.test

```diff
@@ -0,0 +1,113 @@
+--source include/wait_condition.inc
+
+# make sure ibtmp1 is not encrypted
+
+--let $grep_pattern= Lorem ipsum dolor sit amet
+--let $grep_file= $MYSQL_DATA_DIR/ibtmp1
+--let $grep_output= boolean
+--source include/grep_pattern.inc
+
+SET GLOBAL innodb_temp_tablespace_encrypt = ON;
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t02 (a INT) ENGINE=InnoDB ENCRYPTION='Y';
+
+CREATE TEMPORARY TABLE t03 (a TEXT) ENGINE=InnoDB ENCRYPTION='N';
+INSERT INTO t03 VALUES ('Curabitur laoreet, velit non interdum venenatis');
+
+# wait until pages flushed
+--let $wait_condition= SELECT COUNT(*) = 0 FROM performance_schema.global_status WHERE LOWER(variable_name) = 'innodb_buffer_pool_pages_dirty' AND variable_value > 0
+--source include/wait_condition.inc
+
+# make sure ibtmp1 is encrypted now
+
+--let $grep_pattern= Curabitur laoreet, velit non interdum venenatis
+--let $grep_file= $MYSQL_DATA_DIR/ibtmp1
+--let $grep_output= boolean
+--source include/grep_pattern.inc
+
+# this table created in separate file per table tablespace
+CREATE TEMPORARY TABLE t04 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED;
+INSERT INTO t04 VALUES (1), (2), (3);
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t05 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='y';
+
+CREATE TEMPORARY TABLE t06 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='n';
+INSERT INTO t06 VALUES (1), (2), (3);
+
+# test that we can turn encryption OFF and ON
+
+SET GLOBAL innodb_temp_tablespace_encrypt = OFF;
+
+CREATE TEMPORARY TABLE t07 (a INT) ENGINE=InnoDB;
+INSERT INTO t07 VALUES (1), (2), (3);
+
+SET GLOBAL innodb_temp_tablespace_encrypt = ON;
+
+CREATE TABLE t10 (a INT AUTO_INCREMENT PRIMARY KEY, b INT);
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+
+INSERT INTO t10 (b) SELECT b FROM t10;
+INSERT INTO t10 (b) SELECT b FROM t10;
+INSERT INTO t10 (b) SELECT b FROM t10;
+INSERT INTO t10 (b) SELECT b FROM t10;
```

Comment:
What is the purpose of making t10 big?

Reply:
It needs to be big enough so that CREATE INDEX uses temp table